src/tokenize.cpp

```diff
@@ -682,7 +682,7 @@ static bool parse_number(tok_ctx &ctx, chunk_t &pc)
          pc_temp.str.append(ch);
       }
       pc_length = pc_temp.len();
-      ch        = pc_temp.str[pc_length - 1];
+      ch        = (pc_length > 0) ? pc_temp.str[pc_length - 1] : -1;
```

Comment:
How so ? Without my commit the type of  pc_length is int, pc_temp.len(); returns a size_t which can be 0. In the while loop CharTable::IsKw2 can fail right from the beginning.